client/trino-client/src/main/java/io/trino/client/JsonResultRows.java

```diff
@@ -73,22 +54,27 @@ private static class RowWiseIterator
         private final JsonParser parser;
         private final TypeDecoder[] decoders;
 
-        public RowWiseIterator(InputStream stream, TypeDecoder[] decoders)
+        public RowWiseIterator(JsonParser parser, TypeDecoder[] decoders)
```

Comment:
drop this `implements Iterable` abomination

Reply:
Good point!